utils/comparable_bytes.cc

```diff
@@ -15,12 +15,31 @@
 
 logging::logger cblogger("comparable_bytes");
 
+// Encode/Decode the given signed fixed-length integer into byte comparable format.
+// To encode, invert the sign bit so that negative numbers are ordered before the positive ones.
+template <std::signed_integral T>
+static void convert_signed_fixed_length_integer(managed_bytes_view& src, bytes_ostream& out) {
+    constexpr auto sign_mask = T(1) << (sizeof(T) * 8 - 1);
```

Comment:
I expect debug mode will complain about positive value converted to negative. Better use std::make_unsigned_t for all intermediates.